tee_supplicant/src/ree_fs.c

```diff
@@ -47,37 +94,46 @@ static int tee_fs_open(size_t num_params, struct tee_param *params,
 	}
 	strncat(path, name, PATH_MAX);
 
-	file = k_malloc(sizeof(*file));
-	if (!file) {
-		return TEEC_ERROR_GENERIC;
-	}
-
-	fs_file_t_init(file);
-	fd = z_alloc_fd(file, NULL);
+	fd = internal_open(path, flags);
 	if (fd < 0) {
-		rc = TEEC_ERROR_GENERIC;
-		goto free;
-	}
-
-	rc = fs_open(file, path, flags);
-	if (rc < 0) {
-		if (rc == -ENOENT) {
-			rc = TEEC_ERROR_ITEM_NOT_FOUND;
-			goto free;
+		if (flags & FS_O_CREATE) {
+			char *dir;
+
+			dir = dname(path);
+			if (!dir) {
+				rc = TEEC_ERROR_GENERIC;
+				goto out;
+			}
+			rc = fs_mkdir(dir);
+			if (rc < 0) {
+				rc = TEEC_ERROR_GENERIC;
+				goto out;
+			}
+			strcpy(path, REE_FS_MP);
+			strncat(path, name, PATH_MAX);
+			fd = internal_open(path, flags);
```

Comment:
We should handle /1/2/3/file here. Or if it's impossible - just check path to be ./file or ./1/file. 
If directory is well known why don't create it  on init?

Reply:
optee doesn't use directories. This is necessary to create directory for secure storage. Create code which will not be used is not good idea.